R/add.R

```diff
@@ -57,6 +57,17 @@ add_row <- function(.data, ..., .before = NULL, .after = NULL) {
     out <- rbind(.data[indexes, ], df, .data[-indexes, ])
   }
 
+  # If .data is empty, coerce the out column class to .data class
+  if (nrow(.data) == 0) {
+    for (i in 1:ncol(out)) {
```

Comment:
If we go that route, I'd prefer an `rbind_xxx()` function that takes care of the details. For a 0-row data frame you can still query the first row, this gets you a row of typed NAs:

```r
iris[0,][1,] %>% lapply(class)
```

Then, in this specific corner case, we might be able to bind a row of NA values with the new data, and remove the first row again.